src/App.js

```diff
@@ -1,8 +1,72 @@
+import { React, useState, useReducer, useEffect } from "react";
+import { FaBookReader } from "react-icons/fa";
+
+import InputContainer from "./containers/InputContainer";
+import ListItemContainer from "./containers/ListItemContainer";
+
+//로컬 스토리지의 정보구성 {list: {item1, item2, ...}}
+
+const initList = localStorage.getItem("list")
+  ? JSON.parse(localStorage.getItem("list"))
+  : [];
+
+const listReducer = (state, action) => {
+  let listBuffer = [];
+  //DELETE = 아이템 삭제, ADD = 아이템 추가, MODIFY = 아이템 type 수정(done/todo)
+  switch (action.type) {
+    case "DELETE":
+      listBuffer = state.filter((item) => item.id !== action.payload.id);
+      break;
+    case "ADD":
+      listBuffer = state.concat([action.payload]);
+      break;
+    case "MODIFY":
+      listBuffer = state.map((item) =>
+        item.id === action.payload.id ? action.payload : item
+      );
+      break;
```

Comment:
리듀서는 `state`를 기준으로 새로운 값을 리턴하는 방법이 좋습니다. 
```suggestion
  //DELETE = 아이템 삭제, ADD = 아이템 추가, MODIFY = 아이템 type 수정(done/todo)
  switch (action.type) {
    case "DELETE":
      return state.filter((item) => item.id !== action.payload.id);
    case "ADD":
      return [...state, action.payload]
    case "MODIFY":
      return state.map((item) =>
        item.id === action.payload.id ? action.payload : item
      );
```

Reply:
맞습니다, Reducer 안에서 LocalStorage 접근을 동시에 하다보니 이상한 문법이 만들어진 것 같아요. Reducer 사용 시 state 변경 외에 state에 기반한 부가적인 작업을 해도 괜찮은지 궁금하기도 합니다.

Reply:
리듀서 내에서 상태 변경 외의 로직을 쓰는것은 바람직한 패턴이 아닌 것으로 알고 있습니다. 제가 리덕스를 사용해본 적이 없어 리덕스를 통해 상태관리를 할 때 대헌님께서 물어보신 부분을 어떻게 해결하는지는 잘 모르지만, 저의 경우에는 custom hook을 사용해 `dispatch`에서 상태변경을 한 다음 비즈니스 로직(예를 들면 비동기)을 더하는 방식으로 구현하는 편 입니다.

Reply:
커스텀 훅을 사용하는 방법이 있군요, 답변 감사드립니다.  Redux도 제가 임의로 사용하던 방식이라 커스텀 훅을 쓰는 방향으로 다음 코드를 작성해봐야겠습니다!